lib/mocha.js

```diff
@@ -471,7 +491,14 @@ Mocha.prototype.run = function(fn) {
   var options = this.options;
   options.files = this.files;
   var runner = new exports.Runner(suite, options.delay);
-  var reporter = new this._reporter(runner, options);
+
+  // For each loaded reporter constructor, create
+  // an instance and initialize it with the runner
+  var reporters = this._reporters.map(function(reporterConfig) {
+    var _reporter = reporterConfig.fn;
+    options.reporterOptions = reporterConfig.options;
```

Comment:
@santiagoaguiar Since we support IE7, I'd use `stringify`/`parse` 'til we figure out what we do with polyfills / stuff. We lose support for using `function` as value for options (when using Mocha programmatically), but I don't think anyone is doing that.